### PR TITLE
fix: public site user confirmation issue

### PR DESCRIPTION
**Pull request by YazeedLoonat**
## Issue Overview

This PR addresses #[issue ](https://github.com/housingbayarea/bloom/issues/566)
- [x] This change addresses the issue in full

## Description
There was an issue where when a user is trying to confirm their account on the public site, they were running into a problem where there account was never getting confirmed. This meant that they were never logging in even though they clicked the confirmation link on the email.

This is a fix to resolve that so when following the confirmation link they will be logged in 

## How Can This Be Tested/Reviewed?
head over to the public site ( https://hba-public-dev.netlify.app/ )
attempt to create a new account
get the email 
click the confirmation link
you should now be signed in

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [x] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

**Review by ColinBuyck — COMMENTED** [review]
I remember you mentioning that this issue wasn't reproducible in local development so I'm a bit unsure on how to test this fix. Also, I'm struggling to understand the flow and surprised to see that an undefined error within a catch block is a commonly-used path. Do you mind expanding on the reasoning behind this fix?

**Comment by YazeedLoonat** [comment]
@ColinBuyck for sure:

1) I think my big breakthrough was that we can recreate this error locally. If you tear down your local, and put it back up then attempt to go through the account confirmation process you can get a js error. That js error is that we are trying to pull apart undefined in `ConfirmationModal`'s useEffect around line 49. 

2) What that error is coming from is in `AuthContext.ts` we are probably trying to dispatch the `confirmAccount` action before userService is fully instantiated. So we end up returning undefined on line 337. 

This would cause us to fall into the .catch of ConfirmationModal's useEffect, but there isn't an actual error. Thats what was causing the issue. Basically there is a race case between the front end trying to fire off the action and then userService actually being ready to perform that action

hopefully that makes some sense, but lemme know if not

File: sites/public/src/ConfirmationModal.tsx [hunk 1]
```diff
@@ -57,13 +57,15 @@ const ConfirmationModal = (props: ConfirmationModalProps) => {
...
         })
         .catch((error) => {
-          const {
-            response: { data },
-          } = error
-          if (data.statusCode === 406) {
-            setSiteAlertMessage(t(`authentication.createAccount.errors.${data.message}`), "alert")
-          } else {
-            setOpenModal(true)
+          if (!error) {
```
Comment (emilyjablonski) [inline sites/public/src/ConfirmationModal.tsx:60]: I'm feeling a little confused here and could benefit from more color, why would the error be null in the catch, and if its null why are we destructuring it?

**Comment by emilyjablonski** [comment]
Just a heads up I tested this locally, created an account, clicked the confirmation link, and it just took me to the home page & nothing happened - the account didn't get confirmed.

**Review by ColinBuyck — APPROVED** [review]
This worked on my side and looks good to me! 🎬

**Comment by YazeedLoonat** [comment]
Hey @emilyjablonski thats a good question, as it stands right now we already create a few different axios instances that overwrite one another, in this case this one would always get overridden with one once we have the actual accesstoken set up; whether this change goes in or not 

hopefully that makes some sense?

**Review by ludtkemorgan — APPROVED** [review]
This looks good and I'm approving it. I do have two comments though.
1. Is this needed for bloom core as well?
2. I think this is a good sign that the way we are using axios might need some rethinking. Definitely nothing needs to be done now, but something to think of for the future.

**Pull request merged**